lib/puppetserver/ca/utils/http_client.rb

```diff
@@ -195,6 +195,10 @@ def self.check_server_online(settings, logger)
           rescue Puppetserver::Ca::ConnectionFailed => e
             if e.wrapped.is_a? Errno::ECONNREFUSED
               return false
+            elsif suppress_error
+              logger.inform("Could not determine that Puppet Server is offline, " \
+                "connection check failed with error: #{e.wrapped}.")
+              return false
```

Comment:
Back to one of Molly's original questions:
Since we're now catching this error out in the generate action, and doing something different if `force` is true vs false, maybe we can move this out there too. As in, leave this whole file alone and do all the special logic in the action. With the extra logic and special logging we've added, I think I now agree that it makes more sense to do at the call site.

Reply:
Yeah, gonna restore this file to what it was before any changes.  Gonna have to rewrite the test cases as well. Might take a bit longer 😅